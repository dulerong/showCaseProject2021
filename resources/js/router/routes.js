// route paths
import { routePath } from './constantRoutes'

// layouts
import DefaultLayout from '~layouts/default'
import LoginLayout from '~layouts/login'

// pages
import HomePage from '~pages/HomePage'
import LoginPage from '~pages/LoginPage'
import StatPage from '~pages/StatPage'
import NotFoundPage from '~pages/NotFoundPage'

// import { store } from './store'

export default [
  {
    path: routePath.home.path,
    name: routePath.home.name,
    component: HomePage,
    meta: { layout: DefaultLayout }
  },
  {
    path: routePath.stat.path,
    name: routePath.stat.name,
    component: StatPage,
    meta: { layout: DefaultLayout }
  },
  {
    path: routePath.login.path,
    name: routePath.login.name,
    component: LoginPage,
    meta: { layout: LoginLayout }
  },
  {
    path: '*',
    component: NotFoundPage,
    meta: { layout: DefaultLayout, requiresAuth: true }
  }
]