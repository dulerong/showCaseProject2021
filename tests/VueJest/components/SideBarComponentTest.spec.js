import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'
import routes from '~/js/routes'

import SideBar from '~components/crm/SideBar'

describe('App Component Tests', () => {
  const localVue = createLocalVue()
  localVue.use(VueRouter)

  const router = new VueRouter({ routes })

  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const propsData = {
    drawer: {
      isMini: true
    }
  }

  const mountFunction = options => {
    return mount(SideBar, {
      localVue,
      router,
      vuetify,
      propsData,
      ...options,
    })
  }

  it('App renders properly', () => {
    const wrapper = mountFunction()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  // it('Clicking SideBar tab', async () => {
  //   const wrapper = mountFunction()

  //   expect(wrapper.vm.$route.name).toBe('home')

  //   const button = wrapper.findAll('.v-list-item').at(2)
  //   button.trigger('click')

  //   expect(wrapper.vm.$route.name).toBe('search')
  // })
})
