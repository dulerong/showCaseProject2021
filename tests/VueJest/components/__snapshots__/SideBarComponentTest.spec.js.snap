// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Tests App renders properly 1`] = `
<nav class="v-navigation-drawer v-navigation-drawer--fixed v-navigation-drawer--is-mobile v-navigation-drawer--open v-navigation-drawer--temporary theme--light" style="height: 100vh; top: 0px; transform: translateX(0%); width: 256px;">
  <div class="v-navigation-drawer__content">
    <div class="d-flex justify-center align-center" style="height: 60px;"><img src="/images/logo/ikeda.svg"></div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light" style="border: 2px solid #1976d2;">
    <div class="v-list pa-0 v-sheet theme--light">
      <div class="v-list-group">
        <div tabindex="0" aria-expanded="false" role="button" class="v-list-group__header v-list-item v-list-item--link theme--light">
          <div class="v-list-item__title">顧客管理</div>
          <div class="v-list-item__icon v-list-group__header__append-icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></div>
        </div>
        <transition-stub name="expand-transition">
          <!---->
        </transition-stub>
      </div>
      <div class="v-list-group">
        <div tabindex="0" aria-expanded="false" role="button" class="v-list-group__header v-list-item v-list-item--link theme--light">
          <div class="v-list-item__title">集計管理</div>
          <div class="v-list-item__icon v-list-group__header__append-icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></div>
        </div>
        <transition-stub name="expand-transition">
          <!---->
        </transition-stub>
      </div>
      <div class="v-list-group">
        <div tabindex="0" aria-expanded="false" role="button" class="v-list-group__header v-list-item v-list-item--link theme--light">
          <div class="v-list-item__title">イベント管理</div>
          <div class="v-list-item__icon v-list-group__header__append-icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></div>
        </div>
        <transition-stub name="expand-transition">
          <!---->
        </transition-stub>
      </div>
      <div class="v-list-group">
        <div tabindex="0" aria-expanded="false" role="button" class="v-list-group__header v-list-item v-list-item--link theme--light">
          <div class="v-list-item__title">暫定的（削除予定）</div>
          <div class="v-list-item__icon v-list-group__header__append-icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></div>
        </div>
        <transition-stub name="expand-transition">
          <!---->
        </transition-stub>
      </div>
    </div>
  </div>
  <div class="v-navigation-drawer__border"></div>
</nav>
`;
